### PR TITLE
Add TVOS support for Expo apps

### PR DESCRIPTION
**Pull request by vegaro**
Add TV Os support for expo projects and fixes https://github.com/RevenueCat/react-native-purchases/issues/947

### CODE REVIEW

File: ios/RNPurchases.m [hunk 1]
```diff
@@ -191,11 +191,15 @@ - (dispatch_queue_t)methodQueue {
...
 {
+    #if !TARGET_OS_TV
     if (@available(iOS 14.3, macOS 11.1, macCatalyst 14.3, *)) {
         [RCCommonFunctionality enableAdServicesAttributionTokenCollection];
     } else {
         NSLog(@"[Purchases] Warning: tried to enable AdServices attribution token collection, but it's only available on iOS 14.3 or greater or macOS 11.1 or greater.");
     }
+    #else
+    NSLog(@"[Purchases] Warning: AdServices attribution token collection is not available on tvOS.");
+    #endif
```
Comment (aboedo) [inline ios/RNPurchases.m:202]: I guess it's the only platform that doesn't have it
Screenshot 2024-03-22 at 11 08 11 AM

File: package.json [hunk 2]
```diff
@@ -63,8 +63,8 @@
   "license": "MIT",
   "readmeFilename": "README.md",
   "peerDependencies": {
-    "react": ">= 16.6.3",
-    "react-native": ">= 0.58.2"
+    "react": "*",
+    "react-native": "*"
```
Comment (aboedo) [inline package.json:67]: wonder if this is going to come back to bite us if we don't at least pin to major version, is there a way to do like `16.*`?

**Comment by aboedo** [comment]
looking great so far

File: package.json (hunk 2 above)
Comment (vegaro) [reply package.json]: The error when installing only complains about the version of react native

```
npm ERR! Conflicting peer dependency: react-native@0.73.6
npm ERR! node_modules/react-native
npm ERR!   peer react-native@">= 0.58.2" from react-native-purchases@7.24.0
npm ERR!   node_modules/react-native-purchases
npm ERR!     react-native-purchases@"RevenueCat/react-native-purchases#tvos-deps" from the root project
```

So I am actually going to revert the react dependency change and only change `react-native` to `*`. `0.58` is so old it doesn't even show up in https://reactnative.dev/versions lol. `react-native-purchases-ui` is already pointing to `*`. I am confident this change is fine.

File: ios/RNPurchases.m [hunk 3]
```diff
@@ -236,14 +240,14 @@ - (dispatch_queue_t)methodQueue {
...
-    logUnavailablePresentCodeRedemptionSheet();
-    #else
+    #if TARGET_OS_IOS
     if (@available(iOS 14.0, *)) {
         [RCCommonFunctionality presentCodeRedemptionSheet];
     } else {
         logUnavailablePresentCodeRedemptionSheet();
     }
+    #else
+    logUnavailablePresentCodeRedemptionSheet();
```
[the remark this answers is not in the document]
Comment (aboedo) [reply ios/RNPurchases.m:250]: yep!

image

[the remark this answers is not in the document]
Comment (aboedo) [reply ios/RNPurchases.m:250]: so we should check for both things: 
```objective-c
#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
```

**Review by aboedo — CHANGES_REQUESTED** [review]
Almost ready to go! Just one quick thing

**Comment by aboedo** [comment]
@vegaro bump here, feels like it's _veerrrrry_ close to the finish line

**Comment by vegaro** [comment]
oh @aboedo let me check, this one fell through the cracks lol checking now so we can ship it

File: package.json [hunk 4]
```diff
@@ -64,7 +64,7 @@
   "readmeFilename": "README.md",
   "peerDependencies": {
     "react": ">= 16.6.3",
-    "react-native": ">= 0.58.2"
+    "react-native": "*"
```
Comment (aboedo) [inline package.json:67]: are these still needed?

**Review by aboedo — APPROVED** [review]
final check on whether we did actually need the `*` dependency on RN

File: package.json (hunk 4 above)
Comment (vegaro) [reply package.json:67]: yes this is needed so expo installs the dependency correctly without having to force it. 0.58.2 is pretty old so we should be fine. react-native-purchases-ui also has *

**Pull request merged**